tests/main/views/test_drafts.py

```diff
@@ -56,6 +62,455 @@ def service_count(self):
+                **self.basic_questions_to_copy,
+                'targetFramework': 'g-cloud-7'
+            }),
+            content_type='application/json')
+        assert res.status_code == 201
+
+        post_copy_source_service = Service.query.first()
+        assert post_copy_source_service.copied_to_following_framework is True
+
+    @pytest.mark.parametrize('framework_status', Framework.STATUSES)
+    def test_services_can_not_be_copied_to_a_framework_that_is_not_open(self, framework_status):
+        if framework_status == 'open':
+            return
+
+        self.set_framework_status('g-cloud-7', framework_status)
+
+        res = self.client.put(
+            '/draft-services/copy-from/{}'.format(self.service_id),
+            data=json.dumps({
+                **self.updater_json,
+                **self.basic_questions_to_copy,
+                'targetFramework': 'g-cloud-7'
+            }),
+            content_type='application/json')
+
+        assert res.status_code == 400
+        assert 'Target framework is not open' in res.get_data(as_text=True)
+
+
+class TestCopyPublishedFromFramework(DraftsHelpersMixin):
+    def post_to_copy_published_from_framework(
+        self, source_framework_slug="g-cloud-6", questions_to_copy='default', framework_slug='g-cloud-7'
+    ):
+        data = {
+            **self.updater_json,
+            "sourceFrameworkSlug": source_framework_slug,
+            "supplierId": 1,
+        }
+
+        if questions_to_copy == 'default':
+            data.update(self.basic_questions_to_copy)
+        elif questions_to_copy:
+                data.update(questionsToCopy=questions_to_copy)
+
+        res = self.client.post(
+            f'/draft-services/{framework_slug}/paas/copy-published-from-framework',
+            data=json.dumps(data),
+            content_type="application/json",
+        )
+
+        return res
+
+    def test_should_copy_all_lots_services_to_new_framwork(self):
+        self.setup_dummy_services(5, supplier_id=1, lot_id=2)
+
+        services = Service.query.filter(
+            Service.supplier_id == 1,
+            Service.framework_id == 1,  # G-Cloud 6
+            Service.lot_id == 2,
+        ).all()
+
+        res = self.post_to_copy_published_from_framework()
+        assert res.status_code == 201
+
+        data = json.loads(res.get_data(as_text=True))
+        serialized_drafts = data['services']
+
+        db_drafts = DraftService.query.filter(
+            DraftService.supplier_id == 1,
+            DraftService.framework_id == 4,  # G-Cloud 7
+            DraftService.lot_id == 2,
+        ).all()
+
+        assert len(serialized_drafts) == 5
+        assert len(db_drafts) == 5
+        assert all(service['status'] == 'not-submitted' for service in serialized_drafts)
+        assert all(service['frameworkSlug'] == 'g-cloud-7' for service in serialized_drafts)
+        assert all(not service.get('serviceId') for service in serialized_drafts)
```

Comment:
This test might be more precise as `'serviceId' not in service` (or `service['serviceId'] is None` if we expect the key to be present with a None value).